provider/ovh/ovh.go

```diff
@@ -215,7 +230,7 @@ func (p *OVHProvider) zonesRecords(ctx context.Context) ([]string, []ovhRecord,
 		eg.Go(func() error { return p.records(&ctx, &zone, chRecords) })
 	}
 	if err := eg.Wait(); err != nil {
-		return nil, nil, err
+		return nil, nil, provider.NewSoftError(err)
 	}
 	close(chRecords)
 	for records := range chRecords {
```

Comment:
The line `chRecords := make(chan []ovhRecord, len(zones))` creates a slice of a certain lenght. Is it possible that some of the records are nulls? Is it worth to add a null guard?

```go
for records := range chRecords {
  if records == null {
     continue
  }
}
```

Or it will never happen?

Reply:
I don't think this is going to happen as records are returned from the API over the channel.
BTW, I didn't modify this line in the PR so I would rather prefer letting it as it is.

Reply:
I set the discussion as resolved then :+1: